sphinx/builders/html/__init__.py

```diff
@@ -1221,15 +1221,26 @@ def validate_html_favicon(app: Sphinx, config: Config) -> None:
         config.html_favicon = None  # type: ignore
 
 
+UNSET = object()
+
+
 def migrate_html_add_permalinks(app: Sphinx, config: Config) -> None:
     """Migrate html_add_permalinks to html_permalinks*."""
-    if config.html_add_permalinks:
-        if (isinstance(config.html_add_permalinks, bool) and
-                config.html_add_permalinks is False):
-            config.html_permalinks = False  # type: ignore
-        else:
-            config.html_permalinks_icon = html.escape(config.html_add_permalinks)  # type: ignore  # NOQA
+    html_add_permalinks = config.html_add_permalinks
+    if html_add_permalinks is UNSET:
+        return
+
+    if (
+        html_add_permalinks is None or
+        html_add_permalinks is False or
+        html_add_permalinks == ""
```

Comment:
It seems this is equivalent to `not html_add_permalinks`. How about using it instead?